client/src/components/Common/Header/Header.js

```diff
@@ -1,19 +1,48 @@
-import React from 'react'
+import React, { Component } from 'react'
 import GSGlog from '../../../Images/gsg_logo.jpg'
 import Profile from '../../../Images/Ellipse 2.jpg'
 import './Header.css'
 import { Link } from 'react-router-dom'
+// through withRouter You can get access to the history object’s properties 
+// withRouter will pass updated match,location, and history props to the wrapped component whenever it renders.
+import { withRouter } from 'react-router'
+import axios from 'axios'
 
-function Header () {
-  return (
-    <div className='header'>
-      <Link to={'/'} className='header_link'>
-        <img className='logo' src={GSGlog} alt='logo' />
-      </Link>
-      <Link to = {'/home'}>
-        <img className='profile' src={Profile} alt='profile'/></Link>
-    </div>
-  )
+class Header extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      email: '',
+      userId: '',
+      userName: ''
+    }
+  }
+
+  show = ()=> {
+    const { history } = this.props
+    axios.get('/api/check').then(({ data }) => {
+      const { success, email, userId, userName } = data
+      if (success) {
+        this.setState({  email, userId, userName } , () => {
+          history.push(`/home/${userId}`)
+        })
+      } else {
+        return history.push('/login')
+      }
+    })
+  }
+
+  render () {
+    const { userId } = this.state 
+    return (
+      <div className='header'>
+        <Link to={'/'} className='header_link'>
+          <img className='logo' src={GSGlog} alt='logo' />
+        </Link>
+        <Link to = {{pathname:`/home/${userId}`}}>
+          <img className='profile' src={Profile} alt='profile' onClick={this.show}/></Link></div>
```

Comment:
seems there is some redundancy here. You have a link to the user's home page, but then you have the onClick where you do the redirect manually. If the `onClick` is being used, you should be able to remove the Link and just keep the img.